### PR TITLE
ENH Only query primary DB

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11352

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "require": {
         "php": "^8.1",
         "silverstripe/admin": "^2",
-        "silverstripe/framework": "^5.3",
+        "silverstripe/framework": "^5.4",
```
Comment (ChloeSartorelli) [inline composer.json]: No need to bump - the config will just be unused for earlier versions.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Models/LoginSession.php [hunk 2]
```diff
@@ -101,6 +101,8 @@ class LoginSession extends DataObject
 
     private static bool $anonymize_ip = false;
 
+    private static bool $must_use_primary_db = true;
```
Comment (ChloeSartorelli) [inline src/Models/LoginSession.php:108]: Why? Add a code comment so it's clear why we're setting this to true.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (emteknetnz) [reply composer.json]: Updated

File: src/Models/LoginSession.php (hunk 2 above)
Comment (emteknetnz) [reply src/Models/LoginSession.php:108]: Updated

Comment (ChloeSartorelli) [reply src/Models/LoginSession.php:108]: The comment explains what this config property does - but still doesn't say why we need it to be true for this dataobject specifically. I assume it's something to do with ensuring session data doesn't go out of sync?

Comment (emteknetnz) [reply src/Models/LoginSession.php:108]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM, will merge after framework PR

**Pull request merged**